src/hotspot/share/gc/shenandoah/shenandoahOldGeneration.cpp

```diff
@@ -244,7 +243,12 @@ void ShenandoahOldGeneration::handle_failed_transfer() {
 }
 
 size_t ShenandoahOldGeneration::usage_trigger_threshold() const {
-  size_t result = _live_bytes_after_last_mark + (_live_bytes_after_last_mark * _growth_before_compaction) / FRACTIONAL_DENOMINATOR;
+  size_t threshold_by_relative_growth =
+    _live_bytes_after_last_mark + (_live_bytes_after_last_mark * _growth_percent_before_compaction) / 100;
+  size_t threshold_by_growth_into_percent_remaining = (size_t)
+    (_live_bytes_after_last_mark + ((ShenandoahHeap::heap()->soft_max_capacity() - _live_bytes_after_last_mark)
```

Comment:
I need to protect against underflow here.  It might be that _live_bytes_after_last_mark is greater than soft_max_capacity().